com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.Prepass.cs

```diff
@@ -746,8 +746,15 @@ void RenderDBuffer(RenderGraph renderGraph, HDCamera hdCamera, TextureHandle dec
                 return;
             }
 
-            // We need to copy depth buffer texture if we want to bind it at this stage
-            CopyDepthBufferIfNeeded(renderGraph, hdCamera, ref output);
+            bool canReadBoundDepthBuffer = SystemInfo.graphicsDeviceType == GraphicsDeviceType.PlayStation4 ||
```

Comment:
can't we move this code inside the function CopyDepthBufferIfNeeded, will avoid duplication of code

Reply:
This CopyDepthBufferIfNeeded is not the same CopyDepthBufferIfNeeded used in the main path, so duplication is necessary. Moreover, in some cases we cannot skip the copy based on that (we need to eventually copy, is just that we don't for simply the dbuffer pass, not all copies) .